src/api/controllers/UserController.js

```diff
@@ -32,7 +32,60 @@ const authResponse = (user) => ({
 });
 
 module.exports = {
-  changePassword: (req, res) => {},
+  changePassword: (req, res) => {
+    const previousPassword = req.body.previousPassword;
+    const newPassword = req.body.newPassword;
+
+    if (!req.body.previousPassword) {
+      return res.status(400).json({
+        error: "Field `previousPassword` is required",
+      });
+    }
+
+    if (!req.body.newPassword) {
+      return res.status(400).json({
+        error: "Field `newPassword` is required",
+      });
+    }
+
+    User.findById({ _id: req.userData.userId })
+      .exec()
+      .then((user) => {
+        if (!user) {
+          return res.status(401).json({
```

Comment:
Is this really possible? (since we already have to have a valid token for this request to go through)